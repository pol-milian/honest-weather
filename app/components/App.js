import { injectGlobal } from "styled-components";

const React = require("react");
const ReactRouter = require("react-router-dom");

const BrowserRouter = ReactRouter.BrowserRouter;
const Route = ReactRouter.Route;
const Switch = ReactRouter.Switch;

const Header = require("./Header");
const MainSearch = require("./MainSearch");
const Forecast = require("./Forecast");
const Detailed = require("./Detailed");

require.context("../images", true, /\.svg$/);

injectGlobal`
  * {
    box-sizing: border-box;
  }
  html {
    width: 100%;
    height: 100%;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI",
      Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans",
      "Helvetica Neue", Arial, sans-serif,
      "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
    font-size: 1.2rem;  
    
  }
  body {
    position: relative;
    margin: 0;
    color: #707070;
  }
  a {
    text-decoration: none;
  }
`;

class App extends React.Component {
  render() {
    return (
      <BrowserRouter>
        <div>
          <Route render={Header} />
          <Route exact path="/" render={MainSearch} />
          <Route path="/forecast" component={Forecast} />

          <Route path="/detailed/:city" component={Detailed} />
        </div>
      </BrowserRouter>
    );
  }
}

module.exports = App;
