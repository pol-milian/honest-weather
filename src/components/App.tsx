import React from 'react';
import { createGlobalStyle } from 'styled-components';
import { BrowserRouter as Router, Route, Switch, withRouter } from "react-router-dom";
import styledNormalize from 'styled-normalize'
// import { } from 'styled-components/cssprop'
import * as types from 'styled-components/cssprop'
import Main from './Main'
import Forecast from './Forecast'
import Detailed from './Detailed'
import Header from './Header'


const GlobalStyle = createGlobalStyle`
  ${styledNormalize}
  @import url('https://fonts.googleapis.com/css?family=Poppins&display=swap');
  :root {
    --black: #707070;
    --blue: #247ba0;
    --turquoise: #70c1b3;
    --generalMargin: 10vh auto;
    
  }
  * {
    box-sizing: border-box;
  }
  body {
    max-width: 38rem;
    padding: 2rem;
    margin: 0 auto;
    color: var(--black);
    font-size: 20px;
    font-family: 'Poppins', sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }
  h1 {
    font-size: 48px;
    font-weight: 400;
  }
  h2 {
    font-size: 36px;
  }
  p {
    line-height: 1.3;
  }
  a {
    text-decoration: none;
  }
`

export const LocationDisplay = withRouter(({ location }) => (
  <div data-testid="location-display">{location.pathname}</div>
));

const NoMatch = () => <div>No match</div>



const App = () => {
  return (
    <div>
      <GlobalStyle />
      {/* <Header /> */}
        <Switch>
          <Route path="/" exact component={Main} />
          <Route path="/forecast" component={Forecast} />
          <Route path="/detailed/:city" component={Detailed} />
          <Route component={NoMatch} />
          
        </Switch>
        <LocationDisplay />
    </div>
  );
}

export default App;
