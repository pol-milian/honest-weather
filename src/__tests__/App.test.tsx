import React from 'react';
import App from '../components/App';
import { render, cleanup } from '@testing-library/react';

afterEach(cleanup);

test('renders without crashing', () => {
  const { debug, queryByText } = render(<App />);
  expect(queryByText(/what is the weather like in.../i)).toBeTruthy();
  // debug();
})
