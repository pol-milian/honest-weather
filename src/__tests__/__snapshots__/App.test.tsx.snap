// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<main
  className="sc-ifAKCX ftAxxp"
>
  <h1
    className="sc-bxivhb jLngrx"
  >
    What is the weather like in...
  </h1>
  <div
    className="sc-bdVaJa kHIJVD"
  >
    <input
      className="sc-bwzfXH buIknn"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="your shitty city ðŸ’©"
      required={true}
      type="text"
      value=""
    />
    <button
      className="sc-htpNat jSbjIy"
      onClick={[Function]}
      type="button"
    >
      SEARCH!
    </button>
  </div>
</main>
`;
